src/features/session_trace/aggregate/trace/storage.js

```diff
@@ -164,6 +164,9 @@ export class TraceStorage {
       if (!(typeof val === 'number' && val >= 0)) continue
 
       val = Math.round(val)
+      if (this.parent.timeKeeper && this.parent.timeKeeper.ready && val >= this.parent.timeKeeper.correctedOriginTime) {
```

Comment:
```suggestion
      // fallbacks from the performance.timing API supply values with unix timestamps that must be converted to relative ms from origin
      if (this.parent.timeKeeper && this.parent.timeKeeper.ready && val >= this.parent.timeKeeper.correctedOriginTime) {
```

Reply:
I'm not certain `val >= this.parent.timeKeeper.correctedOriginTime` is always applicable or that every timing "val" is strictly always after the "correctedOriginTime".
I suggest just passing an arg to `storeTiming` that branches in this behavior.